modular/blocksyncer/blocksyncer_options.go

```diff
@@ -1,478 +0,0 @@
-				for ; currHeight <= uint64(latestBlockHeight); currHeight++ {
-					// log.Debugw("enqueueing new block", "height", currHeight)
-					exportQueue <- currHeight
-				}
-			}
-		}
-	}
-}
-
-func (b *BlockSyncerModular) getLatestBlockHeight(ctx context.Context) {
-	for {
-		select {
-		case <-ctx.Done():
-			log.Infof("Receive cancel signal, getLatestBlockHeight routine will stop")
-			return
-		default:
-			{
-				latestBlockHeight, err := b.parserCtx.Node.LatestHeight()
-				if err != nil {
-					log.Errorw("failed to get last block from RPCConfig client",
-						"error", err,
-						"retry interval", config.GetAvgBlockTime())
-					continue
-				}
-				Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Store(latestBlockHeight)
-
-				time.Sleep(time.Second)
-			}
-		}
-	}
-}
-
-func (b *BlockSyncerModular) quickFetchBlockData(startHeight uint64) {
-	count := uint64(b.config.Parser.Workers)
-	cycle := uint64(0)
-	startBlock := uint64(0)
-	endBlock := uint64(0)
-	flag := 0
-
-	for {
-		latestBlockHeightAny := Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Load()
-		latestBlockHeight := latestBlockHeightAny.(int64)
-		if latestBlockHeight == int64(endBlock) {
-			continue
-		}
-		log.Info(count*(cycle+1) + startHeight - 1)
-		log.Info(latestBlockHeight)
-		if latestBlockHeight > int64(count*(cycle+1)+startHeight-1) {
-			startBlock = count*cycle + startHeight
-			endBlock = count*(cycle+1) + startHeight - 1
-			//processedHeight := Cast(b.parserCtx.Indexer).ProcessedHeight
-			//if processedHeight != 0 && startBlock-processedHeight > MaxHeightGapFactor*count {
-			//	log.Infof("processedHeight: %d", processedHeight)
-			//	time.Sleep(time.Second)
-			//	continue
-			//}
-			cycle++
-		} else if flag != 0 {
-			startBlock = endBlock + 1
-			if startBlock > uint64(latestBlockHeight) {
-				startBlock = uint64(latestBlockHeight)
-			}
-			endBlock = uint64(latestBlockHeight)
-		} else {
-			flag = 1
-			startBlock = startHeight
-			endBlock = uint64(latestBlockHeight)
-		}
-
-		b.fetchData(startBlock, endBlock)
-	}
-}
-
-func (b *BlockSyncerModular) fetchData(start, end uint64) {
-	log.Infof("fetch data start:%d end:%d", start, end)
-	if start > end {
-		return
-	}
-	wg := &sync.WaitGroup{}
-	wg.Add(int(end - start + 1))
-	for i := start; i <= end; i++ {
-		go func(height uint64) {
-			defer wg.Done()
-
-			for {
-				block, err := b.parserCtx.Node.Block(int64(height))
-				if err != nil {
-					log.Warnf("failed to get block from node: %s", err)
-					continue
-				}
-
-				events, err := b.parserCtx.Node.BlockResults(int64(height))
-				if err != nil {
-					log.Warnf("failed to get block results from node: %s", err)
-					continue
-				}
-				txs, err := b.parserCtx.Node.Txs(block)
-				if err != nil {
-					log.Warnf("failed to get block results from node: %s", err)
-					continue
-				}
-				heightKey := fmt.Sprintf("%s-%d", b.Name(), height)
-				blockMap.Store(heightKey, block)
```

Comment:
pay attention of the memory usage in case of the worker does not consume the data in time.